src/Interval.js

```diff
@@ -86,6 +87,19 @@ Interval.prototype = {
     }
   },
 
+  // Returns a new Interval that has the same extent as this one, but which is relative
+  // to `that`, an Interval that fully covers this one.
+  relativeTo: function(that, newInputStream) {
+    if (this.inputStream !== that.inputStream) {
+      throw errors.intervalSourcesDontMatch();
+    }
+    assert(this.startIdx >= that.startIdx && this.endIdx <= that.endIdx,
```

Comment:
Why is this assertion not a proper error in `errors`, like `intervalSourcesDontMatch`, above? Is this something that can be caused by the user, or can it only happen as a result of a bug in the Ohm implementation?